src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,297 @@
+/**
+ * @file methods/kernel_svm/kernel_svm_impl.hpp
+ * @author Himanshu Pathak
+ *
+ * Implementation of Kernel SVM.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+#define MLPACK_METHODS_KERNEL_SVM_KERNEL_SVM_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "kernel_svm.hpp"
+
+namespace mlpack {
+namespace svm {
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const double regularization,
+    const bool fitIntercept,
+    const size_t maxIter,
+    const double tol,
+    const KernelType kernel) :
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    kernel(kernel)
+{
+  intercept = 0;
+  Train(data, labels, maxIter, tol);
+}
+
+template <typename MatType, typename KernelType>
+KernelSVM<MatType, KernelType>::KernelSVM(
+    const double regularization,
+    const bool fitIntercept,
+    const KernelType kernel) :
+    regularization(regularization),
+    fitIntercept(fitIntercept),
+    kernel(kernel)
+{
+  intercept = 0;
+}
+
+template<typename MatType, typename KernelType>
+double KernelSVM<MatType, KernelType>::Train(
+    const MatType& data,
+    const arma::Row<size_t>& labels,
+    const size_t maxIter,
+    const double tol)
+{
+  trainCoefficients = arma::Row<int> (data.n_cols);
+
+  // Changing labels values to 1, -1 values provided
+  // by user should 0 and 1.
+  for (size_t i = 0; i < data.n_cols; i++)
+  {
+    if (labels(i) == 0)
+      trainCoefficients(i) = -1;
+    else
+      trainCoefficients(i) = 1;
+  }
+
+  // Intializing variable to calculate alphas.
+  alpha = arma::zeros(1, data.n_cols);
+  size_t count = 0;
+
+  // Vector to store error value.
+  arma::vec E = arma::zeros(data.n_cols, 1);
+  // Eta value.
+  double eta = 0;
+
+  // Variable to store lower bound and higher bound.
+  double L = 0;
+  double H = 0;
+
+  // Storing kernel function values.
+  arma::mat K = arma::mat(data.n_cols, data.n_cols);
+
+  // Pre-compute the Kernel Matrix
+  for (size_t i = 0; i < data.n_cols; i++)
+  {
+    for (size_t j = 0; j < data.n_cols; j++)
+    {
+      K(i, j) = kernel.Evaluate(data.col(i), data.col(j));
+    }
+  }
+
+  // Training starts from here.
+  while (count < maxIter)
+  {
+    size_t changedAlphas = 0;
+    for (size_t i = 0; i < data.n_cols; i++)
+    {
+      // Calculate Ei = f(x(i)) - y(i) using (2).
+      // E(i) = b + sum (X(i, :) * (repmat(alphas.*Y,1,n).*X)') - Y(i);
+      E(i) = intercept + arma::sum(alpha % trainCoefficients % K.col(i).t())
+             - trainCoefficients(i);
```

Comment:
can you try to see the shape sizes of all these ones and see if they match.
Also do we need to use `arma::as_scalar` here after taking the sum?

Reply:
@saksham189 Actually problem is not with this code because error occurs when the execution of all code present inside `train()` is finished.

Reply:
Can you elaborate on the error you are referring to?

Reply:
I added `std::cout` statement everywhere. So if you look at the builds here http://ci.mlpack.org/job/pull-requests%20mlpack%20memory/5870/console every statement inside the `train()` function is executed but as soon as the execution of `train()` complete we are getting a memory error.

Reply:
That is convenient, can you comment out each test except `ConcentricCircleDataset`?

Reply:
Done

Reply:
Can you check why the tests are failing?

Reply:
Hm, at least we don't see a memory issue for this one, but maybe that's because the test failed?

Reply:
By mistake I added a wrong `if` that's why it's failing now everything is fine with this 😄

Reply:
I think the memory build is now fixed? and it's not showing any errors now?

Reply:
Yes I have solved the issue with memory builds

Reply:
This is great!

Reply:
What was the issue @himanshupathak21061998 ?

Reply:
There were multiple problems with one was `trainCoefficient` and also a very dumb mistake of not returning any value in `train()`

Reply:
Awesome work @himanshupathak21061998, glad you got it figured out! :100: